### PR TITLE
Upgrade 19

### PR DESCRIPTION
**Pull request by Abdelkader42**
Upgrade Angular 19

### CODE REVIEW

File: tsconfig.app.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "files": [
     "src/main.ts",
     "src/main.server.ts",
-    "server.ts"
+    "server"
```
Comment (JeanMeche) [inline tsconfig.app.json:13]: This doesn't look right.

File: tsconfig.json [hunk 2]
```diff
@@ -12,7 +12,6 @@
     "esModuleInterop": true,
     "sourceMap": true,
     "declaration": false,
-    "downlevelIteration": true,
     "experimentalDecorators": true,
```
Comment (JeanMeche) [inline tsconfig.json:15]: We could drop `experimentalDecorators` too, as we aren't using `@Inject` in this project.

File: src/app/feature/calendar/dialogs/card-details-dialog/card-details-dialog.component.ts [hunk 3]
```diff
@@ -8,8 +8,12 @@ import {SafePipe} from '../../pipes/safe.pipe';
 
 @Component({
   selector: 'xmas-card-details-dialog',
-  standalone: true,
-  imports: [CommonModule, CalendarCardFactoryDirective, SafePipe, SocialProfileUrlPipe],
+  imports: [
+    CommonModule,
```
Comment (JeanMeche) [inline src/app/feature/calendar/dialogs/card-details-dialog/card-details-dialog.component.ts:12]: Ideally we should import the `CommonModule` but own the directives we use. In this case it looks like we're only pulling `NgTemplateOutletDirective`

File: src/app/feature/calendar/calendar-grid.component.ts [hunk 4]
```diff
@@ -8,20 +8,19 @@ import {ShakeWhenDisabledDirective} from './directives/shake-on-click.directive'
...
-    OpenDetailsDialogDirective,
-    ShakeWhenDisabledDirective,
-    SocialProfileUrlPipe,
-  ],
-  templateUrl: './calendar-grid.component.html',
-  styleUrl: './calendar-grid.component.css',
-  changeDetection: ChangeDetectionStrategy.OnPush,
+    selector: 'xmas-calendar-grid',
+    imports: [
+        CommonModule,
```
Comment (JeanMeche) [inline src/app/feature/calendar/calendar-grid.component.ts:13]: It looks like the CommonModule is unused.

File: package.json [hunk 5]
```diff
@@ -50,6 +51,6 @@
     "karma-jasmine": "~5.1.0",
     "karma-jasmine-html-reporter": "~2.1.0",
     "prettier": "^3.1.0",
-    "typescript": "~5.2.2"
+    "typescript": "~5.5.4"
```
Comment (JeanMeche) [inline package.json:54]: We can go directly to 5.6

File: src/app/feature/calendar/cards/candle/candle.component.ts [hunk 6]
```diff
@@ -3,12 +3,11 @@ import {ChristmasCard} from '../../interfaces/christmas-card';
...
 
 @Component({
-  selector: 'xmas-candle',
-  standalone: true,
-  imports: [],
-  templateUrl: './candle.component.html',
-  styleUrl: './candle.component.css',
-  changeDetection: ChangeDetectionStrategy.OnPush,
+    selector: 'xmas-candle',
+    imports: [],
```
Comment (michalgrzegorczyk-dev) [inline src/app/feature/calendar/cards/candle/candle.component.ts:7]: unused, remove?

File: src/app/app.component.ts [hunk 7]
```diff
@@ -3,10 +3,9 @@ import {RouterOutlet} from '@angular/router';
 import {SnowflakesComponent} from './feature/calendar/core/snowflakes/snowflakes.component';
 
 @Component({
-  selector: 'xmas-root',
-  standalone: true,
-  imports: [RouterOutlet, SnowflakesComponent],
-  templateUrl: './app.component.html',
-  styleUrls: ['./app.component.css'],
+    selector: 'xmas-root',
+    imports: [RouterOutlet, SnowflakesComponent],
```
Comment (michalgrzegorczyk-dev) [inline src/app/app.component.ts:7]: wondering why standalone was removed?

Comment (michalgrzegorczyk-dev) [reply src/app/app.component.ts:7]: here and in other components

File: src/app/feature/calendar/cards/candle/candle.component.ts (hunk 6 above)
Comment (michalgrzegorczyk-dev) [reply src/app/feature/calendar/cards/candle/candle.component.ts:7]: here and in other components

File: src/app/app.component.ts (hunk 7 above)
Comment (aboudard) [reply src/app/app.component.ts:7]: standalone is removed in v19, it's the default mode

File: src/index.html [hunk 8]
```diff
@@ -52,30 +52,33 @@
...
+            loading="lazy"
             style="margin-right: 15px; opacity: 0.8"
             width="132"
             height="44"
           />
         </p>
         <p style="margin-bottom: 4px">
           <img
-            src="https://www.drupal.org/files/styles/grid-4-2x/public/Packt%20Logo1_0.png?itok=p5jrImHT"
+            ngSrc="https://www.drupal.org/files/styles/grid-4-2x/public/Packt%20Logo1_0.png?itok=p5jrImHT"
```
Comment (michalgrzegorczyk-dev) [inline src/index.html:69]: 👍🏿

File: src/app/feature/calendar/dialogs/card-details-dialog/card-details-dialog.component.ts (hunk 3 above)
Comment (aboudard) [reply src/app/feature/calendar/dialogs/card-details-dialog/card-details-dialog.component.ts:12]: I think we should read "should not import the CommonModule but only the directives we use", and I agree 👍

File: src/app/feature/calendar/cards/candle/candle.component.ts (hunk 6 above)
Comment (danielglejzner) [reply src/app/feature/calendar/cards/candle/candle.component.ts:7]: Angular 19 = standalone default :)

File: src/app/app.component.ts (hunk 7 above)
Comment (danielglejzner) [reply src/app/app.component.ts:7]: Angular 19 = standalone default :)

Comment (danielglejzner) [reply src/app/app.component.ts:7]: oh @aboudard beat me to it :D

Comment (aboudard) [reply src/app/app.component.ts:7]: Yeah, I do that, I beat people 🦈

Comment (michalgrzegorczyk-dev) [reply src/app/app.component.ts:7]: damn :D

**Review by aboudard — APPROVED** [review]
Not much to say, except, maybe remove the tests or write some, but honestly, remove them ;)
Remove unsused stuff, but it's actually working locally on my machine, nodejs 20, OK

File: src/app/feature/calendar/core/snowflakes/snowflakes.component.ts [hunk 9]
```diff
@@ -20,12 +20,11 @@ function calcAnimationDelay(flakeIndex: number, totalSnowCount: number) {
...
-  styleUrl: './snowflakes.component.css',
-  changeDetection: ChangeDetectionStrategy.OnPush,
+    selector: 'xmas-snowflakes',
+    imports: [],
+    templateUrl: './snowflakes.component.html',
+    styleUrl: './snowflakes.component.css',
+    changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class SnowflakesComponent {
   @ViewChild('snowArea') snowArea!: ElementRef<HTMLSpanElement>;
```
Comment (michalgrzegorczyk-dev) [inline src/app/feature/calendar/core/snowflakes/snowflakes.component.ts:30]: I would suggest to move old api with new signal api because from v19 it is stable.

Comment (JeanMeche) [reply src/app/feature/calendar/core/snowflakes/snowflakes.component.ts:30]: `ng generate @angular/core:signal` :)

File: src/app/app.component.ts (hunk 7 above)
Comment (michalgrzegorczyk-dev) [reply src/app/app.component.ts:7]: What about pipes? were left with this attribute. should we remove standalone property from pipes too - just to make it cleaner and consistent?

Comment (michalgrzegorczyk-dev) [reply src/app/app.component.ts:7]: e.g. SafePipe, SocialProfileUrlPipe

**Comment by michalgrzegorczyk-dev** [comment]
I know it's not part of this issue, but I would also remove test files because all of them are empty ;)

File: src/index.html [hunk 10]
```diff
@@ -52,30 +52,33 @@
...
+            loading="lazy"
             style="margin-right: 15px; opacity: 0.8"
             width="132"
             height="44"
           />
         </p>
         <p style="margin-bottom: 4px">
           <img
-            src="./assets/content/images/angular-space-footer.png"
+            ngSrc="./assets/content/images/angular-space-footer.png"
```
Comment (michalgrzegorczyk-dev) [inline src/index.html:79]: image
🤔

Comment (michalgrzegorczyk-dev) [reply src/index.html:79]: images not loaded on netlify?

**Review by 9kubczas4 — APPROVED** [review]
Overall LGTM, small nit comments left by others to be adjusted.
